src/webots/gui/WbSimulationView.cpp

```diff
@@ -894,43 +901,36 @@ void WbSimulationView::updatePlayButtons() {
 
   QAction *pause = manager->action(WbAction::PAUSE);
   QAction *realtime = manager->action(WbAction::REAL_TIME);
-  QAction *run = manager->action(WbAction::RUN);
   QAction *fast = manager->action(WbAction::FAST);
 
   mToolBar->removeAction(pause);
   mToolBar->removeAction(realtime);
-  mToolBar->removeAction(run);
   mToolBar->removeAction(fast);
 
   QList<QAction *> actions;
 
   switch (WbSimulationState::instance()->mode()) {
     case WbSimulationState::REALTIME:
-      actions << pause << run << fast;
-      break;
-
-    case WbSimulationState::RUN:
-      actions << realtime << pause << fast;
+      actions << pause << fast;
       break;
 
     case WbSimulationState::FAST:
-      actions << realtime << run << pause;
+      actions << realtime << pause;
       break;
 
     default:  // PAUSE
-      actions << realtime << run << fast;
+      actions << realtime << fast;
       break;
   }
 
+  manager->toggle3DView();
```

Comment:
Are there many instructions from this `updatePlayButtons` that are needed in case the 3D rendering is toggled?
Because at first sight it seems more efficient to split the two methods and emit a different signal from the WbSimulationState instance if the 3D rendering  is toggled but the mode is not changed.
I checked just the first slots connected to the `WbSimulationState::modeChanged` signal but it seems that in most cases only the play mode or only the 3D rendering status is needed but not both so two different signals could be better.

Reply:
Agree. I added the signal, it works fine, but I only afraid that I forgot to connect it to something.

Reply:
Once the new signal is added I can double check that everything is correctly connected.